react/features/notifications/components/AbstractNotificationsContainer.js

```diff
@@ -0,0 +1,155 @@
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate(prevProps: P) {
+        const { _notifications, _showNotifications } = this.props;
+
+        if (_notifications.length) {
+            const notification = _notifications[0];
+            const previousNotification
+                = prevProps._notifications.length
+                    ? prevProps._notifications[0]
+                    : undefined;
+
+            if (notification !== previousNotification) {
+                this._notificationDismissTimeout
+                    && clearTimeout(this._notificationDismissTimeout);
+
+                this._notificationDismissTimeout = null;
+
+                if (_showNotifications && notification) {
+                    const { timeout, uid } = notification;
+
+                    this._notificationDismissTimeout = setTimeout(() => {
+                        // Perform a no-op if a timeout is not specified.
+                        if (Number.isInteger(timeout)) {
+                            this._onDismissed(uid);
+                        }
+                    }, timeout);
+                }
+            }
+        }
+    }
+
+    /**
+     * Clear any dismissal timeout that is still active.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentWillUnmount() {
+        this._notificationDismissTimeout
+            && clearTimeout(this._notificationDismissTimeout);
+    }
+
+    _onDismissed: number => void;
+
+    /**
+     * Emits an action to remove the notification from the redux store so it
+     * stops displaying.
+     *
+     * @param {number} uid - The id of the notification to be removed.
+     * @private
+     * @returns {void}
+     */
+    _onDismissed(uid) {
+        this._notificationDismissTimeout
+            && clearTimeout(this._notificationDismissTimeout);
+        this._notificationDismissTimeout = null;
+
+        this.props.dispatch(hideNotification(uid));
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated NotificationsContainer's
+ * props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _notifications: Array,
+ *     _showNotifications: boolean
+ * }}
+ */
+export function _abstractMapStateToProps(state: Object) {
+    // TODO: Per existing behavior, notifications should not display when an
+    // overlay is visible. This logic for checking overlay display can likely be
+    // simplified.
+    const {
+        connectionEstablished,
+        haveToReload,
+        isMediaPermissionPromptVisible,
+        suspendDetected
+    } = state['features/overlay'];
+    const isAnyOverlayVisible = (connectionEstablished && haveToReload)
+        || isMediaPermissionPromptVisible
+        || suspendDetected
+        || state['features/base/jwt'].calleeInfoVisible;
+
+    const { enabled, notifications } = state['features/notifications'];
+
+    return {
+        _notifications: notifications,
```

Comment:
It is old code, but part of this PR according to @lyubomir was to write a solid base for notifications and "old code" can not be excuse. This flag is part of the abstraction.

I can also imagine use cases for this flag, but the questions is do they exist currently ? If not then I'd say it's better to remove it.

Reply:
Oh, I'm not reasoning not to remove, I just wanna know if it would break something or not. @virtuacoplenny do you know if that flag was necessary for AtlasKit?